components/reddit/reddit.app.js

```diff
@@ -0,0 +1,58 @@
+const axios = require("axios");
+const get = require("lodash.get");
+
+module.exports = {
+  type: "app",
+  app: "reddit",
+  methods: {
+    _accessToken() {
+      return this.$auth.oauth_access_token;
+    },
+    _apiUrl() {
+      return `https://oauth.reddit.com`;
+    },
+    async _makeRequest(opts) {
+      if (!opts.headers) opts.headers = {};
+      opts.headers.authorization = `Bearer ${this._accessToken()}`;
+      opts.headers["user-agent"] = "@PipedreamHQ/pipedream v0.1";
+      const { path } = opts;
+      delete opts.path;
+      opts.url = `${this._apiUrl()}${path[0] === "/" ? "" : "/"}${path}`;
+      return (await axios(opts)).data;
+    },
+    wereLinksPulled(reddit_things) {
```

Comment:
You can simplify this method a bit:

```javascript
const links = get(reddit_things, "data.children");
return links && links.length
```

That should covers all of the case you have enumerated in a slightly more compact way.

Reply:
Yeah, much more classy so to say :-)